plugins/kernel_v2/config/vm.rb

```diff
@@ -601,11 +601,6 @@ def finalize!
             options[:type] = :nfs
           end
 
-          # Ignore NFS on Windows
-          if options[:type] == :nfs && Vagrant::Util::Platform.windows?
-            options.delete(:type)
-          end
```

Comment:
bringing a machine up on windows with nfs will have have an output something like

```
C:\Users\vagrant\test>vagrant up ubuntu
Bringing machine 'ubuntu' up with 'hyperv' provider...
==> ubuntu: Verifying Hyper-V is enabled...
==> ubuntu: Verifying Hyper-V is accessible...
    ubuntu: Configuring the VM...
==> ubuntu: Starting the machine...
==> ubuntu: Waiting for the machine to report its IP address...
    ubuntu: Timeout: 120 seconds
    ubuntu: IP: 192.168.64.237
==> ubuntu: Waiting for machine to boot. This may take a few minutes...
    ubuntu: SSH address: 192.168.64.237:22
    ubuntu: SSH username: vagrant
    ubuntu: SSH auth method: private key
==> ubuntu: Machine booted and ready!
It appears your machine doesn't support NFS, or there is not an
adapter to enable NFS on this machine for Vagrant. Please verify
that `nfsd` is installed on your machine, and try again. If you're
on Windows, NFS isn't supported. If the problem persists, please
contact Vagrant support.
```

should Vagrant be able to identify that a synced folder is not supported before the machine boots as part of validating the config?

Reply:
Yes, I do think that it should be running a precheck prior to this point, but for now this will keep things consistent and working as expected.